Export-Excel.ps1

```diff
@@ -620,27 +623,36 @@
           if it is a data table don't do foreach on it (slow) - put the whole table in and set dates on date columns,
           set things up for the end block, and skip the process block #>
         if ($InputObject -is  [System.Data.DataTable])  {
-            #Change TableName if $TableName is non-empty; don't leave caller with a renamed table!
-            $orginalTableName = $InputObject.TableName
-            if ($TableName) {
-                $InputObject.TableName = $TableName
-            }
-            while ($InputObject.TableName -in $pkg.Workbook.Worksheets.Tables.name) {
-                Write-Warning "Table name $($InputObject.TableName) is not unique, adding '_' to it "
-                $InputObject.TableName += "_"
-            }
-            #Insert as a table, if Tablestyle didn't arrive as a default, or $TableName non-null - even if empty
-            if ($null -ne $TableName -or $PSBoundParameters.ContainsKey("TableStyle")) {
-                $null = $ws.Cells[$row,$StartColumn].LoadFromDataTable($InputObject, (-not $noHeader),$TableStyle )
-                # Workaround for EPPlus not marking the empty row on an empty table as dummy row.
-                if ($InputObject.Rows.Count -eq 0) {
-                    ($ws.Tables | Select-Object -Last 1).TableXml.table.SetAttribute('insertRow', 1)
+            if ($Append) {
+                $row ++
+                $null = $ws.Cells[$row,$StartColumn].LoadFromDataTable($InputObject, $false )
+                if ($TableName) {
+                    Add-ExcelTable -Range $ws.Cells[$ws.Dimension] -TableName $TableName -TableStyle $TableStyle
                 }
             }
-            else {
-                $null = $ws.Cells[$row,$StartColumn].LoadFromDataTable($InputObject, (-not $noHeader) )
+            else  {
+                #Change TableName if $TableName is non-empty; don't leave caller with a renamed table!
+                $orginalTableName = $InputObject.TableName
+                if ($PSBoundParameters.ContainsKey("TableName")) {
```

Comment:
I see you changed here the `if ($TableName)` to `if ($PSBoundParameters.ContainsKey("TableName"))`
This makes this example create auto named table "Table1" instead of a table named "MyName":
``` powershell
$DataTable = [Data.DataTable]::new('MyName')
$null = $DataTable.Columns.Add('ID', [Int32])
$null = $DataTable.Columns.Add('Name')
Export-Excel -InputObject $DataTable -Path $Path -TableName ''
```
Was this intentional?
When using `-TableName ''` I think using the name of the table and only if it's also empty auto generate name is preferable.
What do you think is better?

Reply:
Yes. It was intentional, because $tableName is set elsewhere in the function we only want to over-ride the name on the datatable object with something which came in as a parameter if it is passed as a parameter. 

You can't have blank table name, and really the name should be validated as not null or empty.  I'll change that.
Most code samples I've seen over the years just use 
[Data.DataTable]::new() 
and your example would be better that way with  -tableName "MyName" as part of export excel.  (Unless the datatable was later going to be used by something which needed the name.) 
IF people care about the table name in excel, AND they don't control the name in a returned table AND they two names must match then -tableName $dataTable.Tablename works. So I tend view making the names match as a corner case anyway :-)

Reply:
$DataTable.TableName can't be null it's always a string.
`if ($PSBoundParameters.ContainsKey("TableName"))` over-ride the datatable object name in more cases then `if ($TableName)` as `if ($TableName)` exclude empty string and ContainsKey includes it.
But as you say it's a far edge case so you can leave it like it is doesn't really matter

Reply:
:-) 
That's correct. 
In _now_ mode, table is assumed. If -TableName is 

- **$false** (which can be written -Table:$false like a switch)  table is disabled
- **''** - Excel table is named table1,2,3 etc. 
- **Not specified** - use the DataTable's name if there is one, and table1,2,3 if there is not

In _Other_ mode  table is only created if name or style is specified. 

- **''** - Excel table is named table1,2,3 etc. 
- **Not specified**  (i.e. Table style was passed but Table name was not)- use the DataTable's name if there is one, and table1,2,3 if there is not